suites/tipset/message_dedup.go

```diff
@@ -24,71 +24,81 @@ func TestBlockMessageDeduplication(t *testing.T, factory state.Factories) {
 	t.Run("apply a single BLS message", func(t *testing.T) {
 		td := builder.Build(t)
 		defer td.Complete()
-		blkBuilder := drivers.NewTipSetMessageBuilder(td)
+		tipB := drivers.NewTipSetMessageBuilder(td)
+		blkB := drivers.NewBlockBuilder(td.ExeCtx.Miner)
 
 		sender, _ := td.NewAccountActor(address.SECP256K1, big_spec.NewInt(10_000_000))
 		receiver, _ := td.NewAccountActor(address.SECP256K1, big_spec.Zero())
 
-		blkBuilder.WithTicketCount(1).
-			// send value from sender to receiver
-			WithBLSMessageOk(
-				td.MessageProducer.Transfer(receiver, sender, chain.Nonce(0), chain.Value(big_spec.NewInt(100))),
-			).ApplyAndValidate()
+		tipB.WithBlockBuilder(
+			blkB.WithTicketCount(1).
+				// send value from sender to receiver
+				WithBLSMessageOk(
+					td.MessageProducer.Transfer(receiver, sender, chain.Nonce(0), chain.Value(big_spec.NewInt(100))),
+				),
+		).ApplyAndValidate()
 
 		td.AssertBalance(receiver, big_spec.NewInt(100))
 	})
 
 	t.Run("apply a duplicated BLS message", func(t *testing.T) {
 		td := builder.Build(t)
 		defer td.Complete()
-		blkBuilder := drivers.NewTipSetMessageBuilder(td)
+		tipB := drivers.NewTipSetMessageBuilder(td)
+		blkB := drivers.NewBlockBuilder(td.ExeCtx.Miner)
 
 		sender, _ := td.NewAccountActor(address.SECP256K1, big_spec.NewInt(10_000_000))
 		receiver, _ := td.NewAccountActor(address.SECP256K1, big_spec.Zero())
 
-		blkBuilder.WithTicketCount(1).
-			// duplicate the message
-			WithBLSMessageOk(td.MessageProducer.Transfer(receiver, sender, chain.Nonce(0), chain.Value(big_spec.NewInt(100)))).
-			WithBLSMessageOk(td.MessageProducer.Transfer(receiver, sender, chain.Nonce(0), chain.Value(big_spec.NewInt(100)))).
-			// only should have a single result
-			WithResult(exitcode.Ok, drivers.EmptyReturnValue).
-			ApplyAndValidate()
+		tipB.WithBlockBuilder(
+			blkB.WithTicketCount(1).
+				// duplicate the message
+				WithBLSMessageOk(td.MessageProducer.Transfer(receiver, sender, chain.Nonce(0), chain.Value(big_spec.NewInt(100)))).
+				WithBLSMessageOk(td.MessageProducer.Transfer(receiver, sender, chain.Nonce(0), chain.Value(big_spec.NewInt(100)))).
+				// only should have a single result
+				WithResult(exitcode.Ok, drivers.EmptyReturnValue),
```

Comment:
Nit: the `WithBLSMessageOk` is confusing here. Is it asserting both the messages are ok or not?

Better would be `WithBLSMessageDropped` or something explicit like that, and then no `WithResult` special handling here.